src/lib/intent/editor-layout/editor-layout-move-field.ts

```diff
@@ -0,0 +1,91 @@
+import Intent from '../base-intent'
+import { PlanMessage } from '../../interfaces/plan-message'
+import chalk from 'chalk'
+import { EditorLayoutMoveFieldGroupAction } from '../../action/editor-layout/editor-layout-move-field'
+import { SaveEditorInterfaceAction } from '../../action/editorinterface-save'
+
+export default class EditorLayoutMoveFieldIntent extends Intent {
+  isEditorInterfaceIntent () {
+    return true
+  }
+  isEditorLayoutUpdate () {
+    return true
+  }
+  isGroupable () {
+    return true
+  }
+  groupsWith (other: Intent): boolean {
+    return other.isGroupable()
+      && other.isEditorLayoutUpdate()
+      && this.isSameContentType(other)
+  }
+  endsGroup (): boolean {
+    return false
+  }
+  shouldSave (): boolean {
+    return false
+  }
+  shouldPublish (): boolean {
+    return false
+  }
+
+  getPivotId () {
+    return this.payload.movement.pivot
+  }
+
+  getDirection () {
+    return this.payload.movement.direction
+  }
+
+  toActions () {
+    return [
+      new EditorLayoutMoveFieldGroupAction(
+        this.payload.contentTypeId,
+        this.payload.fieldId,
+        this.getDirection(),
+        this.getPivotId()
+      ),
+      new SaveEditorInterfaceAction(this.payload.contentTypeId)
+    ]
+  }
+
+  toPlanMessage (): PlanMessage {
+
+    const direction = this.getDirection()
+    const pivot = this.getPivotId()
+    let humanizedMovement
+
+    if (direction === 'toTheTopOfFieldGroup') {
```

Comment:
We are plastering these magic strings again and again, can we make an effort to add type safety for them?

Reply:
I narrowed types and added a type assertion up in the call stack